zkevm-circuits/src/evm_circuit/execution/coinbase.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        table::BlockContextFieldTag,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition, Transition::Delta,
+            },
+            Cell, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use bus_mapping::eth_types::ToLittleEndian;
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct CoinbaseGadget<F> {
+    same_context: SameContextGadget<F>,
+    coinbase_address: Cell<F>,
+}
+
+impl<F: FieldExt> ExecutionGadget<F> for CoinbaseGadget<F> {
+    const NAME: &'static str = "COINBASE";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::COINBASE;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let coinbase_address = cb.query_cell();
+
+        // Pop the value from the stack
+        cb.stack_push(coinbase_address.expr());
```

Comment:
This should be push?
```suggestion
        // Push the value to the stack
        cb.stack_push(coinbase_address.expr());
```

Reply:
thanks for catching this, updated!